spring-integration-kafka/src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageSource.java

```diff
@@ -257,11 +271,68 @@ public Collection<TopicPartition> getAssignedPartitions() {
 		return Collections.unmodifiableCollection(this.assignedPartitions);
 	}
 
+	@Override
+	public void setBeanClassLoader(ClassLoader classLoader) {
+		this.classLoader = classLoader;
+	}
+
 	@Override
 	protected void onInit() {
 		if (!StringUtils.hasText(this.consumerProperties.getClientId())) {
 			this.consumerProperties.setClientId(getComponentName());
 		}
+
+		Map<String, Object> props = this.consumerFactory.getConfigurationProperties();
+		Properties kafkaConsumerProperties = this.consumerProperties.getKafkaConsumerProperties();
+		this.checkNullKeyForExceptions =
+				this.consumerProperties.isCheckDeserExWhenKeyNull() ||
+						checkDeserializer(findDeserializerClass(props, kafkaConsumerProperties, false));
+		this.checkNullValueForExceptions =
+				this.consumerProperties.isCheckDeserExWhenValueNull() ||
+						checkDeserializer(findDeserializerClass(props, kafkaConsumerProperties, true));
+	}
+
+	@Nullable
+	private Object findDeserializerClass(Map<String, Object> props, Properties consumerOverrides, boolean isValue) {
```

Comment:
I will move most of the duplicated code to `SK.ErrorHandlingUtils`; then you can use it from there.

Reply:
Sounds good!

Reply:
https://github.com/spring-projects/spring-kafka/pull/2756